cmlreaders/readers/eeg.py

```diff
@@ -340,24 +356,27 @@ def as_timeseries(self, epochs: Optional[List[Tuple[float, float]]] = None,
         reader = reader_class(filename=eeg_filename,
                               dtype=dtype,
                               epochs=epochs)  # TODO: channels
-        data = reader.read()
+        data, contacts = reader.read()
 
         if scheme is not None:
             if not reader.rereferencing_possible:
                 raise RereferencingNotPossibleError
-            data = self.rereference(data, scheme)
+            data = self.rereference(data, contacts, scheme)
 
-        # TODO: channels, tstart
-        ts = TimeSeries(data, sample_rate, epochs=epochs)
+        # TODO: tstart
+        ts = TimeSeries(data, sample_rate, epochs=epochs, contacts=contacts)
         return ts
 
-    def rereference(self, data: np.ndarray, scheme: pd.DataFrame) -> np.ndarray:
+    def rereference(self, data: np.ndarray, contacts: List[int],
```

Comment:
Is the assumption here that you always read before you re-reference? If that's the case, then the user gets the list of contacts from the call to read, which they can then pass to rereference, so this is all good. We should probably just be clear in the documentation that you won't be able to do something like reader.load_eeg().rereference(...)

Reply:
Yes. The notes section in the docstring says that you should be using this by passing `scheme` rather than trying to call `rereference` directly. I think this will handle at least 95% of cases where someone wants to use a particular reference scheme. For the other 5%, it makes more sense to me to add a method to `TimeSeries`.